### ISSUE TITLE
MEDIUM: Per station support of virtual categories when using a common MM log

### ISSUE THREAD

**Issue by AaronJowitt** [mairlist/rebelreq2018#19]
If a common MM log is shared among multiple stations, ability to nominate over-ride categories on a per station basis.  i.e. if MM schedules a cat called LOCAL, then Station 1 change that Cat to (user defined) LOCAL1, Station2 changes that cat to LOCAL2.  This is useful for minor changes that are station unique.  Eg – different ID’s, or a local VT (wheras the rest of the VT’s in an hour maybe shared/network)
--

**Comment by tweibert** [comment]
Don't really understand. As far as I understand, the MM log file contains **songs** (and lognotes), but not categories. Even if the category was transferred as part of a log entry, how exactly would mAirList use this information?

**Comment by AaronJowitt** [comment]
With WO (And some other systems), a MM DB must contain two categories per song card.  A) The internal MM category.  B) The automation category.  So we have dozens of different MM categories for music (80A, 80B etc), and we set all those default in MM to set a WO category of MUS.  95% of our MM DB is music, but we also schedule ID's, interviews, WO workflows (Command macros) etc.  In most cases the WO category we assign in MM goes straight through to WO unchanged.  i.e. if MM scheduled a WO cat of MUS, it always goes through to WO a s category MUS.  

But we share a common music log with different stations, and that's where WO virtual categories comes in, to allow us localise ID's, VT's on those stations, even though the source MM is common and shared.  

For example, when we have a MM category called IDL.  (ID Local)   In it, we have items that are scheduled with a WO category of LOC (Local).  When 5 station in WO ingest the same playlist from MM, has a lookup/transposing table for WO categories. WO only supports one virtual category per station, but that does us ok. When Station A import a LOC category event, it flips the category to BCQ.  When station 2 ingests the same music log, it flips the category to BGC.  

WO Assets ID's are not globally unique, they are unique to category.  Assume in MM cat ID1 we schedule LOC/1234.  This is a WO cat of LOC, and an asset ID of 1234. (file name 1234.wav).  In WO when station1 ingest teh log, that ID becomes BCQ/1234, being WO asset 1234.wav from WO category BCQ.  When station 2 imports its log, it becomes BGC/1234, being WO asset 1234.wav from WO category BGC.  That way, the two stations play a localised ID for their market.

We can do same for VT's (local VT's), and commands.  i.e. CMD/BB22 may be a command telling it to route live satellite new to air.  The commands can then be 'localised', so each station routes the audio to via own dedicated Axia mixer, etc.

I'm fully open to a different way to do it if MAL support something similar, as long as its as easy to use as the above.  I can send you more detail on how this WO feature work if you need.

**Comment by tweibert** [comment]
mAirList has two kind of IDs: An "internal" ID which is unique over the entire database (actually the primary key of the items table in the SQL database), and an External ID field which is basically free text and which is not necessarily unique. But it is not filled automatically unlike the Internal ID field.

When interacting with a 3rd party scheduling system, you must decide which IDs to use, internal (enter mAirList IDs in MM) or external (enter MM IDs in mAirList as External IDs).

If you are going to work with Internal IDs, it is impossible to do it the WO way because of the uniqueness of the Internal IDs.

So you would have to switch to External IDs altogether probably. And then maybe use the category as a postfix? Hmmmm.....

**Comment by AaronJowitt** [comment]
Yes.  I think it makes more sense to do that, and we only use external iD's.   We'd eventually (sometime after deployment) suggest;

-  'Auto ID suggest' (when recording a new asset in the system it defaults to suggest the next available free external ID)  which of course can be changed manually if needed 

- ability to 'hide' the internal ID in the GUI, and hide the scheduler windows (perhaps on a user basis..) simply to avoid staff confusion, and declutter/clean up unused features, tabs ID' etc.  Then it would look to staff traditionally more like a system that is reliant on external schedulers.  :-)
